src/chain/shared_state.rs

```diff
@@ -292,3 +321,66 @@ impl Debug for SectionProofChain {
         )
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{chain::SectionInfo, BlsPublicKey, FullId, Prefix, XorName};
+    use std::collections::BTreeSet;
+    use unwrap::unwrap;
+
+    fn gen_section_info(pfx: Prefix<XorName>) -> SectionInfo {
+        let sec_size = 5;
+        let mut members = BTreeSet::new();
+        for _ in 0..sec_size {
+            let id = FullId::within_range(&pfx.range_inclusive());
+            let _ = members.insert(*id.public_id());
+        }
+        unwrap!(SectionInfo::new(members, pfx, None))
+    }
+
+    fn gen_pk(pfx: Prefix<XorName>) -> BlsPublicKey {
+        BlsPublicKey::from_section_info(&gen_section_info(pfx))
+    }
+
+    #[test]
+    fn test_update_their_keys() {
+        let empty: Prefix<XorName> = Default::default();
+        let pfx_0 = empty.pushed(false);
+        let pfx_1 = empty.pushed(true);
+        let pfx_10 = pfx_1.pushed(false);
+        let pfx_11 = pfx_1.pushed(true);
+        let pfx_100 = pfx_10.pushed(false);
+        let pfx_101 = pfx_10.pushed(true);
+
+        // our section is 0
+        let sec_0 = gen_section_info(pfx_0);
+        let mut state = SharedState::new(sec_0);
+
+        // add keys for neighbours 10 and 11
+        let pk_sec_10 = gen_pk(pfx_10);
+        let pk_sec_11 = gen_pk(pfx_11);
+        state.update_their_keys(pfx_10, pk_sec_10.clone());
+        state.update_their_keys(pfx_11, pk_sec_11.clone());
+
+        assert_eq!(state.get_their_key(&pfx_10), Some(&pk_sec_10));
```

Comment:
I think I'd probably use an interface that give the full content of the map.

something like:
`assert_eq!(state.get_their_key(), vec![(pfx_10, pk_sec_10), (pfx_11, pk_sec_11)]);`

Reply:
This would need to make sure that there is a predefined order of the prefixes, but it could be done - probably even easier as a part of the pattern from your other comment :+1: